File: composeApp/src/desktopMain/kotlin/fr/thomasbernard03/androidtools/domain/usecases/logcat/ClearLogcatUseCase.kt
package fr.thomasbernard03.androidtools.domain.usecases.logcat

import fr.thomasbernard03.androidtools.domain.repositories.LogcatRepository
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import org.koin.java.KoinJavaComponent.get

class ClearLogcatUseCase(
    private val logcatRepository: LogcatRepository = get(LogcatRepository::class.java)
) {
    suspend operator fun invoke()  = withContext(Dispatchers.IO) {
        logcatRepository.clearLogcat()
    }
}

File: lib/domain/usecases/logcat/clear_logcat_usecase.dart
import 'package:android_tools/domain/repositories/logcat_repository.dart';

class ClearLogcatUsecase {
  final LogcatRepository logcatRepository;
  const ClearLogcatUsecase({required this.logcatRepository});

  Future<void> call(String deviceId) {
    return logcatRepository.clearLogcat(deviceId);
  }
}
